src/hw-2/task.js

```diff
@@ -0,0 +1,104 @@
+import express from 'express';
+import {
+  getAutoSuggestUsers,
+  createUser, updateUser,
+  deleteUser,
+  getUser
+} from './task.model.js';
+import joi from '@hapi/joi';
+import exprssJoiValidation from 'express-joi-validation';
+
+const app = express();
+const router = express.Router();
+const validator = exprssJoiValidation.createValidator({});
+
+const bodySchema = joi.object({
+  login: joi.string().required(),
+  password: joi.string().alphanum().required(),
+  age: joi.number().integer().min(4).max(130).required()
+});
+
+const paramsSchema = joi.object({
+  id: joi.string().required()
+});
+
+const querySchema = joi.object({
+  loginSubstr: joi.required(),
+  limit: joi.required()
+});
+
+app.use(express.json());
+app.use('/', router);
+app.listen(3000, () => console.log('Listening on port 3000...'));
+
+router.get('/user/:id', async (req, res) => {
+  const id = req.params.id;
+
+  await getUser(id)
+    .then(user => res.json(user))
+    .catch(err => {
+      if (err.status) {
+        res.status(err.status).json({ message: err.message });
+      } else {
+        res.status(500).json({ message: err.message });
+      }
+    });
+});
+
+router.get('/users', validator.query(querySchema), async (req, res) => {
```

Comment:
The validator has to be applied only for create and update operations. It's not necessary to have validation here